source/Halibut/Transport/TcpClientManager.cs

```diff
@@ -53,5 +53,24 @@ public IReadOnlyCollection<TcpClient> GetActiveClients(string thumbprint)
 
             return NoClients;
         }
+
+        public void RemoveDisconnectedClient(TcpClient client)
+        {
+            lock (activeClients)
+            {
+                foreach(var activeClient in activeClients)
```

Comment:
From a clarity point of view, I'm finding the naming `activeClient` harder to understand. Reading it through a few times, it appears to be a `KeyValuePair<Thumbprint, TcpClient>`. I wonder if a simple name change could improve this? `kvp`?